packages/editor-ui/src/components/ApiKeyCreateOrEditModal.vue

```diff
@@ -87,6 +87,17 @@ const props = withDefaults(
 	},
 );
 
+const allFormFieldsAreSet = computed(() => {
+	const isExpirationDateSet =
+		expirationDaysFromNow.value === EXPIRATION_OPTIONS.NO_EXPIRATION ||
+		(expirationDaysFromNow.value === EXPIRATION_OPTIONS.CUSTOM && customExpirationDate.value) ||
+		expirationDate.value;
+
+	return label.value && (props.mode === 'edit' ? true : isExpirationDateSet);
+});
+
+const shouldDisableDate = (date: Date) => Date.now() > date.getTime();
```

Comment:
Maybe this is more descriptive?

```suggestion
const isCustomDateSelectDisabled = (date: Date) => Date.now() > date.getTime();
```